lib/index.js

```diff
@@ -411,9 +382,68 @@ var StateMachine = stampit({
       this.addState(event.to);
     }, this);
 
+    this.mixin = _(this.events)
+      .mapValues(function (event, name) {
+        return this.buildEvent(name);
+      }, this)
+      .assign({
+        can: this.can.bind(this),
+        cannot: this.cannot.bind(this),
+        is: this.is.bind(this),
+        hasState: this.hasState.bind(this),
+        isFinal: this.isFinal.bind(this)
+      })
+      .value();
+
     this.current = this.initial;
-    return this.initTarget(_.first(context.args));
   }
 });
 
-module.exports = StateMachine;
+var Target = stampit({
+  props: {
+    events: {},
+    callbacks: {},
+    initial: 'none',
+    final: null
+  },
+  static: {
+    StateMachine: StateMachine,
+    Promise: global.Promise || require('es6-promise').Promise,
+    FsmError: FsmError
+  },
+  init: function init (context) {
+    var target = _.first(context.args) || this;
+
+    _.defaults(target, {
```

Comment:
this whole thing will prioritize the configuration in the first parameter, but if it's not passed, then it'll try to take it from the second parameter, and if it's not there either, just use the defaults.